core/templates/dev/head/components/SolutionEditorDirective.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the solution editor.
+ */
+
+oppia.directive('solutionEditor', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        solution: '=',
+        objectType: '=',
+        interactionHtml: '=',
+        ruleDescriptionChoices: '=',
+        getOnSaveFn: '&onSave'
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/solution_editor_directive.html'),
+      controller: [
+        '$scope', 'editabilityService', 'stateSolutionService',
+        function($scope, editabilityService, stateSolutionService) {
+          $scope.isEditable = editabilityService.isEditable();
+
+          $scope.editSolutionForm = {};
+          $scope.solutionEditorIsOpen = false;
+          $scope.solutionCorrectAnswerHtml = '';
+
+          $scope.getInteractionHtml = function() {
+            if (!$scope.interactionHtml) {
+              $scope.interactionHtml = (
+                $scope.solution.getInteractionHtml($scope.objectType));
+            }
+          };
+
+          $scope.EXPLANATION_FORM_SCHEMA = {
+            type: 'html',
+            ui_config: {}
+          };
+
+          $scope.openSolutionEditor = function() {
+            if ($scope.isEditable) {
+              $scope.solutionEditorIsOpen = true;
+              $scope.getInteractionHtml();
+            }
+          };
+
+          $scope.submitAnswer = function(answer) {
+            // This function sets correctAnswer. interactionHtml calls this
+            // function when an answer is input.
+            $scope.solution.setCorrectAnswer(answer);
+          };
+
+          $scope.saveThisSolution = function() {
+            $scope.solutionEditorIsOpen = false;
+            $scope.getOnSaveFn()();
+          };
+
+          $scope.cancelThisSolutionEdit = function() {
```

Comment:
Can you either name this cancelThisSolutionEditor or cancelThisSolution? Just to be consistent.